joblib/_parallel_backends.py

```diff
@@ -553,21 +610,24 @@ def effective_n_jobs(self, n_jobs):
 
     def apply_async(self, func, callback=None):
         """Schedule a func to be run"""
-        future = self._workers.submit(SafeFunction(func))
-        future.get = functools.partial(self.wrap_future_result, future)
+        future = self._workers.submit(func)
         if callback is not None:
             future.add_done_callback(callback)
         return future
 
-    @staticmethod
-    def wrap_future_result(future, timeout=None):
-        """Wrapper for Future.result to implement the same behaviour as
-        AsyncResults.get from multiprocessing."""
+    def fetch_result_callback(self, out):
         try:
-            return future.result(timeout=timeout)
-        except CfTimeoutError as e:
-            raise TimeoutError from e
+            return out.result()
```

Comment:
There is no notion of timeout anymore, not sure where this is now handle?

Reply:
For backends that support retrieving in the callback, the timeout doesn't rely on the backend implementation anymore, it's done directly in `BatchCompletionCallback.get_status`.

For other backends, it's same as before.